SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -829,6 +841,50 @@ - (void)videoServiceProtocolDidUpdateFromOldProtocol:(nullable SDLProtocol *)old
     }
 }
 
+#pragma mark - Background Task
```

Comment:
I'm going back and forth on whether this is code bloat. I think this code (and the property) should be in a new class that only manages the background task. It can be then held as a property by the lifecycle manager and started and stopped when the lifecycle manager is started and stopped.

Reply:
Made new class called `SDLBackgroundTaskManager`